src/hooks/useArticles.js

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const useArticles = (shouldSort = false, searchQuery = '') => {
+  const [articles, setArticles] = useState([]);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const queryParams = new URLSearchParams();
+      if (shouldSort) {
+        queryParams.append('sortBy', 'contentLength');
+      }
+      if (searchQuery) {
+        queryParams.append('search', searchQuery);
+      }
+
+      const url = `http://localhost:3010/articles?${queryParams.toString()}`;
+      const response = await fetch(url);
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch articles');
+      }
+
+      const articlesData = await response.json();
+      setArticles(articlesData);
+    } catch (error) {
+      console.error(error);
+    }
+  }, [shouldSort, searchQuery]);
+
+  useEffect(() => {
+    fetchData();
+  }, [fetchData]);
+
+  return { articles, setArticles };
```

Comment:
Don't return the `setArticles` function. By doing that, you're giving too much control over the state of the articles to some other parts of your code.

For example, if you want to let other components to remove articles, create the `removeArticle` function and return it instead of `setArticles`.